### PR TITLE
feat: When Dashboard and APISIX version do not match, make the prompt information clearer (#1944)

### LINKED ISSUE #1944: When Dashboard and APISIX version do not match, make the prompt information clearer
# Feature request

## Please describe your feature

When the APISIX version does not match the Dashboard version, the prompt information is not clear enough to allow users to directly know why the prompt is prompted. The matching version should be given directly

image

## Describe the solution you'd like

Add the current version number and the required version in the prompt message

### PR DESCRIPTION
**Pull request by RedemptionC**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**
provide more information in the prompt message when dashboard version and APISIX don't match

**Related issues**

resolve #1944 

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

File: api/internal/handler/tool/tool.go [hunk 1]
```diff
@@ -101,9 +102,11 @@ func (h *Handler) VersionMatch(c droplet.Context) (interface{}, error) {
 	} else {
 		// TODO: move this to utils
 		return &data.SpecCodeResponse{StatusCode: http.StatusOK, Response: data.Response{
-			Data:    &output,
-			Code:    2000001,
-			Message: "The manager-api and apache apisix are mismatched.",
+			Data: &output,
+			Code: 2000001,
+			Message: fmt.Sprintf("The Dashboard and Apache APISIX are mismatched. "+
+				"The version of Dashboard is %s and should be used with APISIX %s",
```
Comment (tokers) [inline api/internal/handler/tool/tool.go:108]:
```suggestion
				"The version of Dashboard is %s and should be used with Apache APISIX %s",
```

File: api/test/e2enew/version/version_test.go [hunk 2]
```diff
@@ -43,9 +43,11 @@ var _ = ginkgo.Describe("Version", func() {
 			Path:         "/apisix/admin/tool/version_match",
 			Headers:      map[string]string{"Authorization": base.GetToken()},
 			ExpectStatus: http.StatusOK,
-			ExpectBody: []string{"\"code\":2000001",
-				"\"message\":\"The manager-api and apache apisix are mismatched.\"",
-				"\"matched\":false", "apisix_server1", "apisix_server2"},
+			ExpectBody: []string{`"code":2000001`,
+				`The manager-api and apache apisix are mismatched`,
+				`The version of Dashboard is`,
+				`and should be used with APISIX`,
```
Comment (tokers) [inline api/test/e2enew/version/version_test.go:49]:
```suggestion
				`and should be used with Apache APISIX`,
```

**Comment by starsz** [comment]
Hello @RedemptionC,  need to fix the CI.

File: api/internal/handler/tool/tool.go [hunk 3]
```diff
@@ -104,8 +104,8 @@ func (h *Handler) VersionMatch(c droplet.Context) (interface{}, error) {
 		return &data.SpecCodeResponse{StatusCode: http.StatusOK, Response: data.Response{
 			Data: &output,
 			Code: 2000001,
-			Message: fmt.Sprintf("The Dashboard and Apache APISIX are mismatched. "+
-				"The version of Dashboard is %s and should be used with APISIX %s",
+			Message: fmt.Sprintf("The manager-api and apache apisix are mismatched. "+
```
Comment (tokers) [inline api/internal/handler/tool/tool.go:107]: Should use `Manager API` and `Apache APISIX`

**Comment by RedemptionC** [comment]
> Hello @RedemptionC, need to fix the CI.

it seems it's fixed now

**Review by gxthrj — APPROVED** [review]

File: api/internal/handler/tool/tool.go [hunk 4]
```diff
@@ -101,9 +102,11 @@ func (h *Handler) VersionMatch(c droplet.Context) (interface{}, error) {
 	} else {
 		// TODO: move this to utils
 		return &data.SpecCodeResponse{StatusCode: http.StatusOK, Response: data.Response{
-			Data:    &output,
-			Code:    2000001,
-			Message: "The manager-api and apache apisix are mismatched.",
+			Data: &output,
+			Code: 2000001,
+			Message: fmt.Sprintf("The Manager-api and Apache APISIX are mismatched. "+
```
Comment (nic-chen) [inline api/internal/handler/tool/tool.go:107]:
```suggestion
			Message: fmt.Sprintf("The Manager API and Apache APISIX are mismatched. "+
```

File: api/internal/handler/tool/tool.go [hunk 5]
```diff
@@ -101,9 +102,11 @@ func (h *Handler) VersionMatch(c droplet.Context) (interface{}, error) {
...
+				"The version of Dashboard is %s, and should be used with Apache APISIX %s.",
```
Comment (nic-chen) [inline api/internal/handler/tool/tool.go:108]:
```suggestion
				"The version of Manager API is %s, and should be used with Apache APISIX %s.",
```

File: api/internal/handler/tool/tool_test.go [hunk 6]
```diff
@@ -115,7 +115,7 @@ func TestTool_VersionMatch(t *testing.T) {
 						},
 					},
 					Code:    2000001,
-					Message: "The manager-api and apache apisix are mismatched.",
+					Message: "The Manager-api and Apache APISIX are mismatched. The version of Dashboard is , and should be used with Apache APISIX .",
```
Comment (nic-chen) [inline api/internal/handler/tool/tool_test.go:118]:
```suggestion
					Message: "The Manager API and Apache APISIX are mismatched. The version of Manager API is , and should be used with Apache APISIX .",
```

File: api/test/e2enew/version/version_test.go [hunk 7]
```diff
@@ -44,8 +44,9 @@ var _ = ginkgo.Describe("Version", func() {
 			Headers:      map[string]string{"Authorization": base.GetToken()},
 			ExpectStatus: http.StatusOK,
 			ExpectBody: []string{"\"code\":2000001",
-				"\"message\":\"The manager-api and apache apisix are mismatched.\"",
-				"\"matched\":false", "apisix_server1", "apisix_server2"},
+				`"message":"The Manager-api and Apache APISIX are mismatched. ` +
```
Comment (nic-chen) [inline api/test/e2enew/version/version_test.go:47]:
```suggestion
				`"message":"The Manager API and Apache APISIX are mismatched. ` +
```

File: api/test/e2enew/version/version_test.go [hunk 8]
```diff
@@ -44,8 +44,9 @@ var _ = ginkgo.Describe("Version", func() {
...
+					`The version of Dashboard is , and should be used with Apache APISIX ."`,
```
Comment (nic-chen) [inline api/test/e2enew/version/version_test.go:48]:
```suggestion
					`The version of Manager API is , and should be used with Apache APISIX ."`,
```

**Review by iamayushdas — APPROVED** [review]
LGTM

**Review by nic-chen — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**